lib/tests/unit/configuration_test.cc

```diff
@@ -372,6 +376,30 @@ TEST_CASE("Configuration::validate", "[configuration]") {
     }
 }
 
+TEST_CASE("Configuration::validate with unknown config options", "[configuration]") {
+    const char* altArgv[] = {
+    "test-command",
+    "--config-file", UNKNOWN_CONFIG.data(),
```

Comment:
We should use `c_str()` instead of `data()` as it guarantees nul terminated string which is what we want here IMO.

Reply:
Ok.